addons/mail/static/src/models/message_list_view/message_list_view.js

```diff
@@ -6,11 +6,89 @@ import { attr, one } from '@mail/model/model_field';
 registerModel({
     name: 'MessageListView',
     identifyingFields: ['threadViewOwner'],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtBottom() {
+            return (
+                this.scrollTop >= this.scrollHeight - this.clientHeight - this.endThreshold
+            );
+        },
+        /***
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtEnd() {
+            if (this.threadViewOwner.order === 'asc') {
+                return this.isAtBottom;
+            }
+            return this.isAtTop;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtStart() {
+            if (this.threadViewOwner.order === 'asc') {
+                return this.isAtTop;
+            }
+            return this.isAtBottom;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtTop() {
+            return this.scrollTop <= this.endThreshold;
+        },
+    },
     fields: {
+        clientHeight: attr(),
         /**
          * States the OWL component of this message list view
          */
         component: attr(),
+        /**
+         * The margin that we use to detect that the scrollbar is a the end of
+         * the threadView.
+         */
+        endThreshold: attr({
```

Comment:
This is either badly named or badly used.

The way the field is currently used is both "bottom threshold" and "top threshold", but not "end" threshold.

Its usage should probably be moved inside the "end" compute method instead, according to its name? But then if it's only used in one place, it might as well be defined locally as we have no plan to change or access this value elsewhere.

Reply:
My idea was that `end` and `start` would not be dependent of the current order like `top` and `bottom`.
It make it easier to think about the this part of the code if you remove the concern of `top` and `bottom`.
Therefor `endThreshold` look like the best naming since it's the threshold at the `end` regardless of the top/bottom.

It could just be a `const` in each function I guess.